roles/os_zabbix/library/zbxapi.py

```diff
@@ -200,25 +231,55 @@ def __init__(self, parent):
         # User.delete
         # User.get
         for method_name in method_names:
-            meta_method(Class, method_name)
+            meta_method(_class, method_name)
         # Return our subclass with all methods attached
-        return Class
+        return _class
 
 # Attach all ZabbixAPI.classes to ZabbixAPI class through metaprogramming
-for class_name, method_names in ZabbixAPI.classes.items():
-    setattr(ZabbixAPI, class_name, ZabbixAPI.meta(class_name, method_names))
+for _class_name, _method_names in ZabbixAPI.classes.items():
+    setattr(ZabbixAPI, _class_name, ZabbixAPI.meta(_class_name, _method_names))
+
+def exists(content, key='result'):
+    ''' Check if key exists in content or the size of content[key] > 0
+    '''
+    if not content.has_key(key):
+        return False
+
+    if not content[key]:
+        return False
+
+    return True
+
+def diff_content(from_zabbix, from_user):
+    ''' Compare passed in object to results returned from zabbix
+    '''
+    terms = ['search', 'output', 'groups', 'select', 'expand']
+    regex = '(' + '|'.join(terms) + ')'
+    retval = {}
+    for key, value in from_user.items():
+        if re.findall(regex, key):
+            continue
+
+        if from_zabbix[key] != str(value):
+            retval[key] = str(value)
+
+    return retval
 
 def main():
+    '''
+    This main method runs the ZabbixAPI Ansible Module
+    '''
 
     module = AnsibleModule(
-        argument_spec = dict(
+        argument_spec=dict(
             server=dict(default='https://localhost/zabbix/api_jsonrpc.php', type='str'),
             user=dict(default=None, type='str'),
             password=dict(default=None, type='str'),
             zbx_class=dict(choices=ZabbixAPI.classes.keys()),
-            action=dict(default=None, type='str'),
+            #action=dict(default=None, type='str'),
```

Comment:
At this point, I think we should just remove this line. We can add it back if we need it, but I'm hoping we'll never need it.